pyiron_base/jobs/job/extension/executor.py

```diff
@@ -0,0 +1,37 @@
+def get_thread_executor(max_workers):
+    from concurrent.futures import ThreadPoolExecutor
+
+    return ThreadPoolExecutor(max_workers=max_workers)
+
+
+def get_process_executor(max_workers):
+    from concurrent.futures import ProcessPoolExecutor
+
+    return ProcessPoolExecutor(max_workers=max_workers)
+
+
+def get_pympipool_mpi_executor(max_workers):
+    from pympipool.mpi.executor import PyMPIExecutor
+
+    return PyMPIExecutor(max_workers=max_workers)
+
+
+def get_pympipool_slurm_executor(max_workers):
+    from pympipool.slurm.executor import PySlurmExecutor
+
+    return PySlurmExecutor(max_workers=max_workers)
+
+
+def get_pympipool_flux_executor(max_workers):
+    from pympipool.flux.executor import PyFluxExecutor
+
+    return PyFluxExecutor(max_workers=max_workers)
+
+
+EXECUTORDICT = {
+    "ThreadPoolExecutor": get_thread_executor,
+    "ProcessPoolExecutor": get_process_executor,
+    "PyMPIPoolExecutor": get_pympipool_mpi_executor,
+    "PyMPIPoolSlurmExecutor": get_pympipool_slurm_executor,
+    "PyMPIPoolFluxExecutor": get_pympipool_flux_executor,
```

Comment:
# (Minor) UX consistency

It is a bit weird to me that some of these incorporate the module path and some don't. Not too bad for users, but the pattern/intent is unclear. What convention should future devs follow for extending the whitelist? 

Using a dict of functions demands that a choice be made for this at all, and in this particular case the difference in treatment and module capitalization makes it unclear exactly what that choice was.

Reply:
The `pyiron_base/jobs/job/extension/executor.py` is now removed.